compute_tools/src/rsyslog.rs

```diff
@@ -0,0 +1,88 @@
+use std::process::Command;
+use std::{fs::OpenOptions, io::Write};
+
+use anyhow::{Context, Result};
+use tracing::info;
+
+fn get_rsyslog_pid() -> Option<String> {
+    let output = Command::new("pgrep")
+        .arg("rsyslogd")
+        .output()
+        .expect("Failed to execute pgrep");
+
+    if !output.stdout.is_empty() {
+        let pid = std::str::from_utf8(&output.stdout)
+            .expect("Invalid UTF-8 in process output")
+            .trim()
+            .to_string();
+        Some(pid)
+    } else {
+        None
+    }
+}
+
+// Start rsyslogd with the specified configuration file
+// If it is already running, do nothing.
+fn start_rsyslog(rsyslog_conf_path: &str) -> Result<()> {
+    let pid = get_rsyslog_pid();
+    if let Some(pid) = pid {
+        info!("rsyslogd is already running with pid: {}", pid);
+        return Ok(());
+    }
+
+    let _ = Command::new("/usr/sbin/rsyslogd")
+        .arg("-f")
+        .arg(rsyslog_conf_path)
+        .arg("-i")
+        .arg("/var/run/rsyslogd/rsyslogd.pid")
+        .output()
+        .context("Failed to start rsyslogd")?;
+
+    // Check that rsyslogd is running
+    if let Some(pid) = get_rsyslog_pid() {
+        info!("rsyslogd started successfully with pid: {}", pid);
+    } else {
+        return Err(anyhow::anyhow!("Failed to start rsyslogd"));
+    }
+
+    Ok(())
+}
+
+pub fn configure_and_start_rsyslog(
+    log_directory: &str,
+    tag: &str,
+    remote_endpoint: &str,
+) -> Result<()> {
+    let config_content = format!(
+        r#"
+# Load imfile module to read log files
+module(load="imfile")
+
+# Input configuration for log files in the specified directory
+input(type="imfile" File="{}/*.log" Tag="{}" Severity="info" Facility="local0")
+global(workDirectory="/var/log")
+
+# Forward logs to remote syslog server
+*.* @@{}
+"#,
```

Comment:
Can you put this into file and include it with `include_str!`? Similar to how we do with some long queries. That'd improve readability

Reply:
The content of the config file here is being built dynamically based on the function parameters:
we use log_directory, tag and remote_endpoint.

So there is not a lot to move to const str.

It may look long because of comments, but apart from that it's only 2 rows - what files to read and where to send them.

Reply:
You can use results of `include_str!` in `format!`, see example for SQL queries, we substitute some parameters there

Reply:
Let's keep in mind that we will need four rsyslog config states:

1. Disabled
2. HIPAA
3. Logs export
4. HIPAA & logs export <-- maybe not

The resulting rsyslog config may be a concatenation of the building blocks necessary for each state.

E.g (3) will need to enable TCP receiver and a different forwarder.